robot_smach_states/src/robot_smach_states/manipulation/grab.py

```diff
@@ -349,10 +356,47 @@ def __init__(self, robot, item, arm):
                                                 'failed': 'RESET_FAILURE'})
 
             smach.StateMachine.add('GRAB', PickUp(robot, arm, item),
-                                   transitions={'succeeded': 'done',
+                                   transitions={'succeeded': 'GRASP_DETECTOR',
+                                                'failed': 'RESET_FAILURE'})
+
+            smach.StateMachine.add('GRASP_DETECTOR', ActiveGraspDetector(robot, arm),
+                                   transitions={'true': 'done',
+                                                'false': 'SAY_GRASP_NOT_SUCCEEDED',
+                                                'failed': 'done',
+                                                'cannot_determine': 'SAY_GRASP_NOT_SUCCEEDED'})
```

Comment:
Not sure about this transition

Reply:
please be more specific in what you do not like about it.

The transition cannot-determine occurs when the fingers are already too close together to detect a change, which happens when the grasp fails. 
A good next step would be to check the dimensions of the object we are grasping to determine this case ahead of time. But for now I think it is a good idea to keep the changes (relatively) small in this PR